File: Server/FIHS/Models/PesticideModels/Pesticide.cs
using FIHS.Models.PestModels;
using Microsoft.AspNetCore.Mvc.ModelBinding.Validation;
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace FIHS.Models.PesticideModels
{
    public class Pesticide
    {
        public int Id { get; set; }
        [Required]
        public string Name { get; set; }
        public double? Price { get; set; }
        [MaxLength(500)]
        public string? Description { get; set; }
        public string? Manufactuer { get; set; }
        [MaxLength(800)]
        public string? UsageInstructions { get; set; }
        public string? Toxicity { get; set; }
        public string? Type { get; set; }
        public string? ImageURL { get; set; }
        [ValidateNever]
        public virtual IEnumerable<PestsPesticides> Pests { get; set; }

    }

}

    

File: Client/FIHS/models/Pesticide.ts
import * as yup from 'yup'

export const PesticideSchema = yup.object({
    name: yup.string().required('برجاء إدخال الاسم'),
    image: yup.mixed().required('برجاء إرفاق صورة'),
    description: yup.string().required('برجاء إدخال الوصف'),
    manufactuer: yup.string().required('برجاء إدخال المصنع'),
    toxicity: yup.string().required('برجاء إدخال درجة السمية'),
    type: yup.string().required('برجاء إدخال النوع'),
    usageInstructions: yup.string().required('برجاء إدخال طرق الاستعمال')
})
export type Pesticide = yup.InferType<typeof PesticideSchema>